storage/innobase/include/srw_lock.h

```diff
@@ -39,13 +39,25 @@ template<bool spinloop>
 class pthread_mutex_wrapper final
 {
   pthread_mutex_t lock;
+  bool initialized{false};
+
 public:
+  pthread_mutex_wrapper()
+  {
+    init();
+  }
+
   void init()
   {
+    if (initialized)
+      return;
```

Comment:
I would like to try another way of fixing this, that is, adding a missing `init()` call somewhere. It should be a no-op for other mutex implementations. It is better to do things without additional state variables and run-time conditions if possible.

For my experiment ab931d656dde299301e291731e2df89a6c448f9c in https://buildbot.mariadb.org/#/builders/234/builds/31470/steps/8/logs/stdio I got a stack trace that identifies a culprit object:
```
server.dll!pthread_mutex_wrapper<0>::wr_lock()[srw_lock.h:67]
server.dll!log_t::append_prepare<0,0>()[mtr0mtr.cc:1042]
```

Reply:
f7b11442e2f992226ea95026d80f72c9e92821ef is an one-liner fix on top of my instrumentation experiment:
```diff
diff --git a/storage/innobase/log/log0log.cc b/storage/innobase/log/log0log.cc
index a485a875ee1..5396806be82 100644
--- a/storage/innobase/log/log0log.cc
+++ b/storage/innobase/log/log0log.cc
@@ -132,6 +132,7 @@ bool log_t::create()
 #endif
 
   latch.SRW_LOCK_INIT(log_latch_key);
+  lsn_lock.init();
 
   last_checkpoint_lsn= FIRST_LSN;
   log_capacity= 0;
```
It was successfully tested on 4 targets where the tests would previously fail:
* Microsoft Windows ([non-debug](https://buildbot.mariadb.org/#/builders/239/builds/22932) and [debug](https://buildbot.mariadb.org/#/builders/234/builds/31489)); the instrumentation caused these to fail as mentioned above.
* [Apple macOS on ARMv8](https://buildbot.mariadb.org/#/builders/590/builds/7443) (in progress)
* [IBM AIX on POWER](https://buildbot.mariadb.org/#/builders/622/builds/3810) (in progress).

I will force-push this pull request with that fix. @DaveGosselin-MariaDB Thank you for your debugging. If we hadn’t been lucky with my instrumentation on Windows, I would have been at a complete loss.